### ISSUE TITLE
cyral_policy resource creates nill identities when undefined

### ISSUE THREAD

**Issue by igaskin** [cyralinc/terraform-provider-cyral#300]
**Summary**

When creating a [cyral_policy_rule](https://registry.terraform.io/providers/cyralinc/cyral/latest/docs/resources/policy_rule#identities) and omitting the optional `identities` field, the `identitites` field is still created when running `terraform apply`.

The nil `identities` map breaks subsequent queries with the error that would use the cyral policy:
```
ERROR:  Operation blocked due to a policy violation: Policy customer-data violated: read is disallowed
```

It appears that the provider correctly omits this field if it is undefined.
https://github.com/cyralinc/terraform-provider-cyral/blob/30c18a231452631a0166e63fbb162bb3094ed9dd/cyral/resource_cyral_policy_rule.go#L19

_sample terraform_
```
resource "cyral_policy_rule" "customer_data_rule" {
  policy_id = cyral_policy.customer_data.id

 reads {
    data = ["*"]
    rows     = 1
    severity = "medium"
  }
}
```

_generated policy rule_
![Screen Shot 2022-10-06 at 3 24 31 PM](https://user-images.githubusercontent.com/5673097/195209912-946c101d-0ccc-4d2c-92f1-6a9e11404b0a.png)

**Expectation**
When a `cyral_policy_rule` is defined without an `identities` block, there should be no `identities` created in the cyral prolicy

**Comment by wcmjunior** [comment]
Hi @igaskin , we appreciate your time to write this thorough description of the error. We will work to investigate and fix it. I will keep you posted.

**Comment by Yowgf** [comment]
Thanks for reporting the issue @igaskin ! We have resolved it for provider versions `v3.0.1` and `v2.10.2`. The problem was that  the `Identities` field in the `PolicyRule` struct was not a pointer, so the `omitempty` JSON tag essentially doesn't work.
